.github/workflows/changelog.yml

```diff
@@ -1,34 +1,20 @@
-# This is a basic workflow to help you get started with Actions
+name: Check changelog and version are matching
 
-name: Check changelog change
-
-# Controls when the action will run. 
 on:
-  # Triggers the workflow on pull request events but only for the develop branch
   pull_request:
     branches: [ develop ]
 
-# A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
-  # This workflow contains a single job called "build"
   check:
-    # The type of runner that the job will run on
     runs-on: ubuntu-latest
 
-    # Steps represent a sequence of tasks that will be executed as part of the job
     steps:
-      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
-      - uses: actions/checkout@v2
-        with:
-          fetch-depth: 0
-
-      # Runs a single command using the runners shell
-      - name: Check if changelog was changed
+      - uses: actions/checkout@v1
+      - name: Set up Python
+        uses: actions/setup-python@v1
+      - name: Install dependencies
         run: |
-          if [[ $(git diff origin/develop Changelog.md) ]]; then
-            echo "Version was changed"
-            exit 0 
-          else
-            echo "No version change detected"
-            exit 1 
-          fi
+          python3 -m pip install --upgrade pip
+          python3 -m pip install -r requirements_version_check.txt
+      - name: Run PlatformIO
+        run: python3 version_check.py
```

Comment:
pass the target branch to the script. this way we can use this script even for other branches (e.g. future v2.x.x branch)